connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java

```diff
@@ -103,6 +109,52 @@ protected InboundConnectorResult<ProcessInstance> triggerStartEvent(
     }
   }
 
+  protected InboundConnectorResult<CorrelatedMessageStart> triggerMessageStartEvent(
+      InboundConnectorDefinitionImpl definition,
+      MessageStartEventCorrelationPoint correlationPoint,
+      Object variables) {
+
+    // String correlationKey = extractCorrelationKey(correlationPoint, variables);
+
+    if (!isActivationConditionMet(definition, variables)) {
+      LOG.debug("Activation condition didn't match: {}", correlationPoint);
+      return new MessageStartCorrelationResult(
+          correlationPoint.messageName(),
+          new CorrelationErrorData(CorrelationErrorReason.ACTIVATION_CONDITION_NOT_MET));
+    }
+
+    Object extractedVariables = extractVariables(variables, definition);
+
+    try {
+      // fixme
+      PublishMessageResponse result =
+          zeebeClient
+              .newPublishMessageCommand()
+              .messageName(correlationPoint.messageName())
+              .correlationKey("") // must be empty to start new process
+              // https://docs.camunda.io/docs/components/modeler/bpmn/message-events/#message-start-events
+              .tenantId(definition.tenantId())
+              .variables(extractedVariables)
+              .send()
+              .join();
```

Comment:
I think we should support the `messageId` here.

Reply:
Good point. I thought that `id` is something internal to Zeebe, apparently we can set it ourselves. I will add it and test.

Reply:
Yeah, its actually quite important and we need to support it also via the element templates (FEEL expression) as its the only way to achieve idempotency with zeebe.

Reply:
Updated